src/aws-cpp-sdk-core/include/aws/core/utils/event/EventHeader.h

```diff
@@ -299,6 +301,36 @@ namespace Aws
                     return m_eventHeaderVariableLengthValue;
                 }
 
+                inline Aws::String ToString() const
+                {
+                    switch (m_eventHeaderType)
+                    {
+                        case EventHeaderType::BOOL_TRUE:
+                        case EventHeaderType::BOOL_FALSE:
+                            return Utils::StringUtils::to_string(GetEventHeaderValueAsBoolean());
+                        case EventHeaderType::BYTE:
+                            return Utils::StringUtils::to_string(GetEventHeaderValueAsByte());
+                        case EventHeaderType::INT16:
+                            return Utils::StringUtils::to_string(GetEventHeaderValueAsInt16());
+                        case EventHeaderType::INT32:
+                            return Utils::StringUtils::to_string(GetEventHeaderValueAsInt32());
+                        case EventHeaderType::INT64:
+                            return Utils::StringUtils::to_string(GetEventHeaderValueAsInt64());
+                        case EventHeaderType::BYTE_BUF:
+                            return Aws::String(reinterpret_cast<char*>(GetEventHeaderValueAsBytebuf().GetUnderlyingData()), GetEventHeaderValueAsBytebuf().GetLength());
```

Comment:
byte buf provides no guarantees to be null terminated. this seems dangerous

Reply:
That's why the string is constructed from ptr + length, not just null-terminated pointer.

Reply:
oh yeah, you are right, missed the final part. ignore this comment. 
we have internal methods to get a proper c_str from buf internally, but i dont think we expose those in bindings